projects/npm-tools/packages/npm-scripts/src/utils/getMergedConfig.js

```diff
@@ -96,20 +97,30 @@ function getMergedConfig(type, property) {
 
 	switch (type) {
 		case 'babel':
-			mergedConfig = deepMerge(
-				[require('../config/babel'), getUserConfig('babel')],
-				deepMerge.MODE.BABEL
-			);
-
-			// (Temporary) special case required by:
-			//
-			// https://github.com/liferay/liferay-npm-tools/issues/303
-			//
-			// TODO: Remove once incremental-dom is no longer used in
-			// liferay-portal.
-
-			mergedConfig = hackilySupportIncrementalDOM(mergedConfig);
-
+			{
+				const {major, minor} = getDXPVersion() || {};
+
+				const baseConfig =
+					major === undefined ||
+					major > 7 ||
+					(major === 7 && minor > 3)
+						? require('../config/babel')
+						: require('../config/babel-legacy');
+
+				mergedConfig = deepMerge(
+					[baseConfig, getUserConfig('babel')],
+					deepMerge.MODE.BABEL
+				);
+
+				// (Temporary) special case required by:
+				//
+				// https://github.com/liferay/liferay-npm-tools/issues/303
+				//
+				// TODO: Remove once incremental-dom is no longer used in
+				// liferay-portal.
```

Comment:
Seems like something we will be able to skip, at least for 7.4 and above, real soon now...